R/summraize_over_group_and_time_range.R

```diff
@@ -13,58 +13,80 @@ summarize_over_group_and_time_range <- function (data_all, country_last_update_i
     filter(set == "country") |>
     select(-name, -set) |>
     left_join(country_last_update_info, by = "unit") |>
-    filter(!is.na(last_update))
-  
-  # if time_range
-  data_table  <-
-    data_filtered |>
+    filter(!is.na(last_update))|>
     filter(dplyr::between(
       time,
-      as.Date(start_date), 
+      as.Date(start_date),
       as.Date(end_date)
     ))
-  
-  if (grouped) {
-    data_table <-
-      data_table |>
-      rename_with(function(x) "group", {{ category }}) |>
-      filter(!is.na(group)) |>
-      mutate(unit = group, name = group)
-  } else {
-    data_table <-
-      data_table |>
-      mutate(group = unit)
-  }
-  
-  data_table <-
-    data_table |>
+# browser()
+  data_summarized <-
+    data_filtered |>
     group_by(unit) |>
     summarize(
       name = name[1],
-      group = group[1],
-      pop_100k               = if(grouped) sum(pop_100k[time == end_date], na.rm = TRUE) else mean(pop_100k, na.rm = TRUE),
-      pop                    = if(grouped) sum(pop[time == end_date], na.rm = TRUE) else mean(pop, na.rm = TRUE),
-      avg_cap_new_cases      = mean(all_new_cases / pop       , na.rm = TRUE),
-      sum_cap_new_cases      = sum(all_new_cases / pop        , na.rm = TRUE),
-      avg_cap_new_deaths     = mean(all_new_deaths / pop      , na.rm = TRUE),
-      sum_cap_new_deaths     = sum(all_new_deaths / pop       , na.rm = TRUE),
-      avg_cap_new_tests      = mean(all_new_tests / pop       , na.rm = TRUE),
-      sum_cap_new_tests      = sum(all_new_tests / pop        , na.rm = TRUE),
-      avg_all_new_cases      = mean(all_new_cases             , na.rm = TRUE),
-      sum_all_new_cases      = sum(all_new_cases              , na.rm = TRUE),
-      avg_all_new_deaths     = mean(all_new_deaths            , na.rm = TRUE),
-      sum_all_new_deaths     = sum(all_new_deaths             , na.rm = TRUE),
-      avg_all_new_tests      = mean(all_new_tests             , na.rm = TRUE),
-      sum_all_new_tests      = sum(all_new_tests              , na.rm = TRUE),
-      avg_pos                = avg_all_new_cases / avg_all_new_tests,
-      avg_cap100k_new_cases  = mean(all_new_cases / pop_100k  , na.rm = TRUE),
-      sum_cap100k_new_cases  = sum(all_new_cases / pop_100k   , na.rm = TRUE),
-      avg_cap100k_new_deaths = mean(all_new_deaths / pop_100k , na.rm = TRUE),
-      sum_cap100k_new_deaths = sum(all_new_deaths / pop_100k  , na.rm = TRUE),
-      avg_cap100k_new_tests  = mean(all_new_tests / pop_100k  , na.rm = TRUE),
-      sum_cap100k_new_tests  = sum(all_new_tests / pop_100k   , na.rm = TRUE),
+      # group = group[1],
+      pop_100k                 = mean(pop_100k, na.rm = TRUE),
```

Comment:
Why do we need to have a mean for the population?
The population of each country is unique, so it should be used in the group_by